pkg/alerts/alertmanager/types.go

```diff
@@ -0,0 +1,95 @@
+package alertmanager
+
+import (
+	"fmt"
+	"net/http"
+)
+
+var (
+	silenceStates = map[string]bool{"expired": true, "active": true, "pending": true}
+)
+
+// Client is the object of the alert manager client
+type Client struct {
+	Token   string       `json:"token" yaml:"token"`
+	HostURL string       `json:"hostUrl" yaml:"hostUrl"`
+	Client  *http.Client `json:"client" yaml:"client"`
+}
+
+// StatusEndpoint is the status enpoint of the alert manager client
+type StatusEndpoint struct {
+	Client  `json:"alertmanagerClient" yaml:"alertmanagerClient"`
+	FullURL string `json:"fullUrl" yaml:"fullUrl"`
+}
+
+// SilencesEndpoint is the silences enpoint of the alert manager client
+type SilencesEndpoint struct {
```

Comment:
Isn't that the same structure?

If you still wish to use a not complete url and keep the different data types (did you think of interfaces?), you could then use a variable in such structure for relative endpoint, which could be "status" for StatusEndpoint. This could be filled by default with tags. It would improve readability IMO, but less than including the whole mumbojumbo in the Get().

... On top of that, I prefer the endpoint being the full URL to be honest :) I avoid the temptation to guess.